bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudAccountBridgeHandler.java

```diff
@@ -161,9 +162,9 @@ public void refreshData() {
                 }
 
                 logger.debug("iCloud bridge data refresh complete.");
-            } catch (NumberFormatException e) {
-                logger.warn("iCloud returned an incorrect account status code : '{}'",
-                        iCloudData.getICloudAccountStatusCode());
+            } catch (NumberFormatException | JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Response = " + json);
```

Comment:
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "iCloud response invalid: " + e.getMessage());
```

And remove the logger. Thing status changes are logged anyway